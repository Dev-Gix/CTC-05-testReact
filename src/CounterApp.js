import React,{useState} from 'react';
import PropTypes from 'prop-types';


const CounterApp = ({value = 10})=>{

    const [counter,setCounter] = useState(value);

    const handleadd = ()=>{setCounter(counter + 1)};

    const handleaddReset = ()=>{setCounter(value)};

    const handleSubstract = ()=>{setCounter(counter - 1)};

    return (
    <>
    <h1>CounterApp</h1>
    <h2>{counter}</h2>
    <div className="botones">
    <button className="btn-incrementar" onClick={handleadd}>+1</button>
    <button className="btn-Reiniciar" onClick={handleaddReset}>Reiniciar</button>
    <button className="btn-decrementar" onClick={handleSubstract}>-1</button>
    </div>
    </>
    );
}

CounterApp.propTypes ={
    value: PropTypes.number
};

export default CounterApp;