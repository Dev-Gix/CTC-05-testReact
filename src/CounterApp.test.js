import React from 'react'
import {shallow} from 'enzyme';
import CounterApp from './CounterApp';

describe('Prueba de CounterApp',()=>{
    test('Match con snapshot ', () => {
      const  wrapper = shallow(<CounterApp />);
        expect(wrapper).toMatchSnapshot();
       
    });
    test('Debe de mostrar el valor por defecto de 100 ', () => {
        const  wrapper = shallow(<CounterApp />);
          expect(wrapper).toMatchSnapshot();
         
      });

      test('deve de mostrar el valor por defecto de 100 ',()=>{

        const  wrapper = shallow(<CounterApp value={100}/>);
        const counterText=wrapper.find('h2').text().trim();
        expect(counterText).toBe('100');

      })
    
})