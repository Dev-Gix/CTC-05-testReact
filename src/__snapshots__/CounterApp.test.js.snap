// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Prueba de CounterApp Debe de mostrar el valor por defecto de 100  1`] = `
<Fragment>
  <h1>
    CounterApp
  </h1>
  <h2>
    10
  </h2>
  <div
    className="botones"
  >
    <button
      className="btn-incrementar"
      onClick={[Function]}
    >
      +1
    </button>
    <button
      className="btn-Reiniciar"
      onClick={[Function]}
    >
      Reiniciar
    </button>
    <button
      className="btn-decrementar"
      onClick={[Function]}
    >
      -1
    </button>
  </div>
</Fragment>
`;

exports[`Prueba de CounterApp Match con snapshot  1`] = `
<Fragment>
  <h1>
    CounterApp
  </h1>
  <h2>
    10
  </h2>
  <div
    className="botones"
  >
    <button
      className="btn-incrementar"
      onClick={[Function]}
    >
      +1
    </button>
    <button
      className="btn-Reiniciar"
      onClick={[Function]}
    >
      Reiniciar
    </button>
    <button
      className="btn-decrementar"
      onClick={[Function]}
    >
      -1
    </button>
  </div>
</Fragment>
`;
